components/brave_sync/crypto/crypto.cc

```diff
@@ -151,7 +151,7 @@ bool PassphraseToBytes32(const std::string& passphrase,
   bytes->resize(DEFAULT_SEED_SIZE);
   if (bip39_mnemonic_to_bytes(nullptr, passphrase.c_str(), bytes->data(),
                               bytes->size(), &written) != WALLY_OK) {
-    LOG(ERROR) << "bip39_mnemonic_to_bytes failed";
+    CHECK(false) << "bip39_mnemonic_to_bytes failed";
```

Comment:
I think you meant to put this in `PassphraseFromBytes32`
The change you did here will crash the browser when UI pass a invalid mnemonic, and UI relies on it to tell users the sync words are invalid.

Reply:
@darkdh , yes, thanks, I will revert this because you are 100% right

Reply:
force-pushed at https://github.com/brave/brave-core/compare/c91ceb642e45dcfbbdf82e945b54eb5032dc10a1..b8715733f07249e053329251c7681db46182af1b